### PR TITLE
SWC-5396: set global app state to isEditing when new react evaluation editor page is shown.

### PR DESCRIPTION
**Pull request by jay-hodgson**
also check global editor state on entity tab change

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/ChallengeTab.java [hunk 1]
```diff
@@ -94,6 +91,7 @@ private void showEvaluationEditor(String entityId, String evaluationId){
 		// a different Evaluation editor
 
 		EvaluationEditorReactComponentPage evaluationEditor = ginInjector.createEvaluationEditorReactComponentPage();
+		globalApplicationState.setIsEditing(true);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/ChallengeTab.java:94]: flip editing to true before showing the editor.

File: src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/ChallengeTab.java [hunk 2]
```diff
@@ -102,6 +100,7 @@ private void showEvaluationEditor(String entityId, String evaluationId){
 					evaluationEditor.removeFromParent();
 					view.showAdminTabContents();
 					evaluationList.refresh();
+					globalApplicationState.setIsEditing(false);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/ChallengeTab.java:103]: and back to false on callback.

File: src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/Tab.java [hunk 3]
```diff
@@ -143,6 +148,18 @@ public void addTabClickedCallback(CallbackP<Tab> onClickCallback) {
 
 	@Override
 	public void onTabClicked() {
+		if (globalAppState.isEditing()) {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/Tab.java:151]: on any entity tab click, check if we are currently in edit mode.  if so, confirm with the user that changes may be lost.

File: src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/Tabs.java [hunk 4]
```diff
@@ -15,11 +17,8 @@
 	@Inject
 	public Tabs(TabsView view) {
 		this.view = view;
-		onTabClickedCallback = new CallbackP<Tab>() {
-			@Override
-			public void invoke(Tab tab) {
-				onTabClicked(tab);
-			}
+		onTabClickedCallback = tab -> {
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/Tabs.java:20]: no real change here

File: src/test/java/org/sagebionetworks/web/unitclient/widget/entity/tabs/ChallengeTabTest.java [hunk 5]
```diff
@@ -50,20 +61,35 @@
 	CookieProvider mockCookieProvider;
 	ChallengeTab tab;
 	@Mock
-	AuthenticationController authenticationController;
+	AuthenticationController mockAuthenticationController;
 	@Mock
-	GlobalApplicationState globalApplicationState;
+	GlobalApplicationState mockGlobalApplicationState;
+	@Mock
+	EvaluationEditorReactComponentPage mockEvaluationEditorReactComponentPage;
```
Comment (jay-hodgson) [inline src/test/java/org/sagebionetworks/web/unitclient/widget/entity/tabs/ChallengeTabTest.java:68]: test that the evaluation editor react component page is configured and that the global editing state is flipped to true, and verify global editing state is flipped to false on callback.

**Review by zimingd — APPROVED** [review]

**Pull request merged**